### PR TITLE
ref: Use simple modulo #shards to get the shard number

### PR DESCRIPTION
**Pull request by olksdr**
As a follow up for #1454 we decided that we do not have to hash the org id to get the shard number and simple modulo #shard will be sufficient.

As part of this small refactoring the `unwrap` was also removed and Result is returned instead, to propagate the error to the caller if it happens.

#skip-changelog

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]

File: relay-server/src/actors/store.rs [hunk 1]
```diff
@@ -143,23 +143,19 @@ pub struct ShardedProducer {
...
+    pub fn get_producer(&self, sharding_key: u64) -> Result<(&str, &ThreadedProducer), StoreError> {
+        let shard = sharding_key % self.shards;
         let (topic_name, producer) = self
             .producers
             .iter()
             .take_while(|(k, _)| *k <= &shard)
             .last()
             .map(|(_, v)| v)
-            .expect("At least one shard is defined");
+            .ok_or(StoreError::InvalidKafkaShardRange)?;
```
Comment (untitaker) [inline relay-server/src/actors/store.rs:156]: this is an unrelated change right? we don't think this is now more likely

**Review by untitaker — APPROVED** [review]

**Review by jan-auer — APPROVED** [review]

File: relay-server/src/actors/store.rs (hunk 1 above)
Comment (olksdr) [reply relay-server/src/actors/store.rs:156]: Yes, this change is unrelated. And error should not happen. 
But I also wanted to remove any `unwrap`/`expect` since I already make changes to this function.
